addons/account/models/account_move_line.py

```diff
@@ -456,7 +456,7 @@ def _compute_currency_id(self):
             else:
                 line.currency_id = line.currency_id or line.company_id.currency_id
 
-    @api.depends('product_id', 'journal_id')
+    @api.depends('product_id')
```

Comment:
```
Have 2 Customer invoices journals
Create an invoice
Add a line and edit product description
Change journal
```

Sadly, same, I can't reproduce consistently.

I managed to do it **ONCE** only by coming from the `sale.order`. When I change the journal for the first time, it changes what was written by the `sale.order` but if I change what is written, it doesn't change anymore when switching journals.

Reply:
Thanks for checking!
I realized the invoice needs to be saved once, I updated the commit message